File: backend/app/schemas/verification.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from app.models.enums import DocumentType, DocumentStatus


class PresignRequest(BaseModel):
    file_name: str
    file_type: str
    document_type: DocumentType


class PresignResponse(BaseModel):
    presigned_url: str
    file_url: str


class DocumentSubmit(BaseModel):
    file_url: str
    document_type: DocumentType


class VerificationDocumentResponse(BaseModel):
    id: int
    user_id: int
    type: DocumentType
    file_url: str
    status: DocumentStatus
    reviewer_id: Optional[int] = None
    notes: Optional[str] = None
    llm_verified: Optional[bool] = None
    llm_confidence: Optional[float] = None
    llm_recommendation: Optional[str] = None
    llm_reasoning: Optional[str] = None
    llm_issues: Optional[list[str]] = None
    llm_extracted_info: Optional[dict] = None
    llm_verified_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True


class VerificationStatusResponse(BaseModel):
    national_id: Optional[VerificationDocumentResponse] = None
    contract: Optional[VerificationDocumentResponse] = None
    user_status: str
    can_post: bool


class DocumentReview(BaseModel):
    notes: Optional[str] = None

File: packages/shared/src/schemas/verification.ts
import { z } from "zod";

export const DocumentTypeSchema = z.enum(["NATIONAL_ID", "CONTRACT"]);

export const DocumentStatusSchema = z.enum([
  "PENDING",
  "APPROVED",
  "REJECTED",
  "REQUEST_MORE_DETAILS",
]);

export const VerificationDocumentSchema = z.object({
  id: z.number(),
  user_id: z.number(),
  compound_id: z.number().nullable().optional(),
  type: DocumentTypeSchema,
  file_url: z.string(),
  status: DocumentStatusSchema,
  reviewer_id: z.number().nullable(),
  notes: z.string().nullable(),
  llm_verified: z.boolean().nullable(),
  llm_confidence: z.number().nullable(),
  llm_recommendation: z.string().nullable(),
  llm_reasoning: z.string().nullable(),
  llm_issues: z.array(z.string()).nullable(),
  llm_extracted_info: z.record(z.any()).nullable(),
  llm_verified_at: z.string().datetime().nullable(),
  created_at: z.string().datetime(),
});

export const VerificationStatusResponseSchema = z.object({
  national_id: VerificationDocumentSchema.nullable(),
  contract: VerificationDocumentSchema.nullable(),
  user_status: z.string(),
  can_post: z.boolean(),
  compound_id: z.number().nullable().optional(),
  compound_name: z.string().nullable().optional(),
});

export const PresignRequestSchema = z.object({
  file_name: z.string(),
  file_type: z.string(),
  document_type: DocumentTypeSchema,
});

export const PresignResponseSchema = z.object({
  presigned_url: z.string(),
  file_url: z.string(),
});

export const DocumentSubmitSchema = z.object({
  file_url: z.string(),
  document_type: DocumentTypeSchema,
});

export type VerificationDocument = z.infer<typeof VerificationDocumentSchema>;
export type VerificationStatusResponse = z.infer<typeof VerificationStatusResponseSchema>;
export type PresignRequest = z.infer<typeof PresignRequestSchema>;
export type PresignResponse = z.infer<typeof PresignResponseSchema>;
export type DocumentSubmit = z.infer<typeof DocumentSubmitSchema>;
export type DocumentType = z.infer<typeof DocumentTypeSchema>;
export type DocumentStatus = z.infer<typeof DocumentStatusSchema>;

